src/header.rs

```diff
@@ -677,7 +677,17 @@ impl Header {
     /// this header.
     pub fn set_cksum(&mut self) {
         let cksum = self.calculate_cksum();
-        octal_into(&mut self.as_old_mut().cksum, cksum);
+
+        let dst = &mut self.as_old_mut().cksum;
+        let o = format!("{:o}", cksum);
```

Comment:
This has historically been really finnicky, but this change seems fine to me! Perhaps though the implementation could still use `octal_into`, but the slice passed in could be two elements shorter which are manually initialized here? (avoids the need to duplicate some of the implementation)

Reply:
yep I thought about pulling it all into `octal_into` but I thought it was a bit cryptic so I kept it separate (so you could more clearly see what was going on) - I'll update `octal_into` as you recommend.